src/components/edit_account/EditAccountSettings.js

```diff
@@ -0,0 +1,432 @@
+            || !this.state.newPasswordReentryIsValid) {
+            this.setState({
+                'currentPassword': '',
+                'newPassword': '',
+                'newPasswordReentry': ''
+            });
+            return;
+        }
+    }
+
+    /**
+     * Revalidate email inputs. If there is undefined or false states, set re-entered email inputs to empty
+     * @param {Event} event 
+     */
+    handleEmailChangeSubmit(event) {
+        event.preventDefault();
+
+        let passwordVerify = this.validateCurrentPassword(this.state.emailPasswordVerify);
+        
+        this.handleSetState('newEmailIsValid',
+            this.validateEmail());
+        this.handleSetState('newEmailReentryIsValid', 
+            this.validateEmailReentry());
+
+        if (!passwordVerify || !this.state.newEmailIsValid
+            || !this.state.newEmailReentryIsValid) {
+            this.setState({
+                newEmail: '',
+                newEmailReentry: '',
+                emailPasswordVerify: ''
+            });
+            return;
+        }
+    }
+
+    /**
+     * Updates the component state when the user types in an input field.
+     * @param {Event} event
+     */
+    handleInputChange(event) {
+        // Make sure the value matches the requirements for its field
+        const value = event.target.value;
+        switch (event.target.name) {
+        case 'newPassword':
+            if (this.state.newPasswordIsValid !== undefined) {
+                this.setState({
+                    'newPasswordIsValid': this.validatePassword(value)
+                });
+            } else {
+                this.setState({
+                    'newPasswordIsValid': undefined
+                });
+            }
+            break;
+        case 'newPasswordReentry':
+            this.setState({
+                'newPasswordReentryIsValid': undefined
+            });
+            break;
+        case 'newEmail':
+            if (value.length === 0 || this.state.newEmailIsValid === undefined){
+                this.setState({
+                    'newEmailIsValid': undefined
+                });
+            } else {
+                this.setState({
+                    'newEmailIsValid': this.validateEmail(value)
+                });
+            }
+            break;
+        case 'newEmailReentry':
+            if (value.length === 0){
+                this.setState({
+                    'newEmailReentryIsValid': undefined
+                });
+            } else if (this.validateEmailReentry(value)){
+                this.setState({
+                    'newEmailReentryIsValid': true
+                });
+            } else {
+                this.setState({
+                    'newEmailReentryIsValid': undefined
+                });
+            }
+            break;
+        default:
+            break;
+        }
+        this.setState({
+            [event.target.name]: value,
+        });
```

Comment:
Again, you are using `setState` too many times in one function. You can probably just call it once at the end here.